features/util.py

```diff
@@ -100,8 +102,54 @@ def wait_for_boot(container_name: str) -> None:
         except ValueError:
             print("Unexpected runlevel output: ", process.stdout.strip())
             runlevel = None
-        if runlevel == "2":
+        if runlevel in ("2", "5"):
             break
         time.sleep(sleep_time)
     else:
         raise Exception("System did not boot in {}s".format(sum(retries)))
+
+
+def lxc_get_series(name: str, image: bool = False):
+    """Check series name of either an image or a container.
+
+    :param name:
+        The name of the container or the image to check its series.
+    :param image:
+        If image==True will check image series
+        If image==False it will check container configuration to get series.
+
+    :return:
+        The series of the container or the image.
+       `None` if it could not detect it (
+           some images doesn't have this field into properties).
+    """
+
+    if not image:
+        output = subprocess.check_output(
+            ["lxc", "config", "get", name, "image.release"],
+            universal_newlines=True,
+        )
+        series = output.rstrip()
+        return series
+    else:
+        image_output = "image_output.yaml"
+        with open(image_output, "w") as fileoutput:
+            subprocess.run(["lxc", "image", "show", name], stdout=fileoutput)
+        with open(image_output, "r") as fileoutput:
+            prop = yaml.safe_load(fileoutput)
+            print(" `lxc image show` output: ", prop)
+        fileoutput.close()
+        os.remove(image_output)
+        try:
+            series = prop["properties"]["release"]
+            print(
+                """\n You are providing a %s image.
```

Comment:
thoughout uaclient code, please try to use the string .format method  instead of %s. For example print("\n You are providing a  {series} image...".format(series=series))

Reply:
ok. I've changed it